### PR TITLE
Bug 1212874 - Losing connection while syncing disables Firefox Sync. …

### PR DESCRIPTION
**Pull request by ferjm**
…r=michielbdejong,yifan

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=79e6f2618bb5af9f26f8936fc7ba529e6fd410e6

File: tv_apps/smart-system/test/unit/sync_manager_test.js [hunk 1]
```diff
@@ -654,6 +674,24 @@ suite('smart-system/SyncManager >', () => {
...
+    const ERROR_DIALOG_CLOSED_BY_USER = 'fxsync-error-dialog-closed-by-user';
+    const ERROR_GET_FXA_ASSERTION = 'fxsync-error-get-fxa-assertion';
+    const ERROR_INVALID_SYNC_ACCOUNT = 'fxsync-error-invalid-account';
+    const ERROR_OFFLINE = 'fxsync-error-offline';
+    const ERROR_REQUEST_SYNC_REGISTRATION =
+      'fxsync-error-request-fxsync-registration';
+    const ERROR_SYNC_INVALID_REQUEST_OPTIONS =
+      'fxsync-error-invalid-request-options';
+    const ERROR_SYNC_REQUEST = 'fxsync-error-request-failed';
+    const ERROR_UNKNOWN = 'fxsync-error-unknown';
```
Comment (begeeben) [inline tv_apps/smart-system/test/unit/sync_manager_test.js]: Can we use the ERRORs required from line 24?

Comment (ferjm) [inline tv_apps/smart-system/test/unit/sync_manager_test.js]: Unfortunately, I couldn't make that work. They are not available on the suite global scope, only within the tests and *setup/*teardown functions.

File: apps/system/js/sync_manager.js [hunk 2]
```diff
@@ -746,8 +762,15 @@
 
     user: {
       set: function(user) {
-        asyncStorage.setItem(SYNC_USER, user, () => {
-          this.store.set(SYNC_USER, user);
+        // We need to check if there's an user logged in during the enabled
```
Comment (michielbdejong) [inline apps/system/js/sync_manager.js]: Nit: an user -> a user. Phonetically, "yoozer" :) starts with a consonant, and that's what decides "a"/"an"

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=67765f88587ebab69cf7db1a96b98090a3aaa34e

**Pull request merged**